packages/flutter_tools/test/src/context.dart

```diff
@@ -125,9 +125,21 @@ void testUsingContext(
           Analytics: () => NoOpAnalytics(),
         },
         body: () {
-          return runZonedGuarded<Future<dynamic>>(() {
+          // runZonedGuarded has a scary documenation that says:
+          //
+          // "returning a future created inside the zone, and waiting for it
+          // outside of the zone, will risk the future not being seen to
+          // complete."
+          //
+          // for example, see our own test suite:
+          // https://github.com/flutter/flutter/blob/5987563e4aecb34fca446ea804943bb8d27d8fcd/packages/flutter_tools/test/general.shard/base/async_guard_test.dart#L279-L306
+          //
+          // so, we create a completer outside the zone, and complete it inside
+          // the zone in baiscally every terminal condition.
```

Comment:
sp baiscally

I would suggest, but when I do the CLA bot gets confused

Reply:
Done.